odoo/addons/base/models/res_lang.py

```diff
@@ -209,57 +232,69 @@ def install_lang(self):
                 partner.write({'lang': lang_code})
         return True
 
-    @tools.ormcache('code')
-    def _lang_get_id(self, code):
-        return self.with_context(active_test=True).search([('code', '=', code)]).id
-
-    @tools.ormcache('code')
-    def _lang_get_direction(self, code):
-        return self.with_context(active_test=True).search([('code', '=', code)]).direction
-
-    @tools.ormcache('url_code')
-    def _lang_get_code(self, url_code):
-        return self.with_context(active_test=True).search([('url_code', '=', url_code)]).code or url_code
+    # ------------------------------------------------------------
+    # cached methods for **active** languages
+    # ------------------------------------------------------------
+    @property
+    def CACHED_FIELDS(self) -> OrderedSet:
+        """ Return the fields to cache for the active languages
+        Please promise all these fields don't depend on other models and context
+        and are not translated.
+        """
+        return OrderedSet(['id', 'name', 'code', 'iso_code', 'url_code', 'active', 'direction', 'date_format',
+                           'time_format', 'week_start', 'grouping', 'decimal_point', 'thousands_sep', 'flag_image_url'])
+
+    def _get_data(self, **kwargs: Any) -> LangData:
+        """ Get the language data for the **first** given field value in kwargs
+        For example, get_data(code='en_US') will return the LangData
+        for the res.lang record whose 'code' field value is 'en_US'
+
+        :param dict kwargs: {field_name: field_value, ...}
+                field_name is the first key in kwargs and in self.CACHED_FIELDS
+                Try to reuse used 'field_name's: 'id', 'code', 'url_code'
+        :return: Valid LangData if (field_name, field_value) pair is for an
+                **active** language. Otherwise, Dummy LangData which will return
+                `False` for any field
+        :rtype: LangData
+        :raise: UserError if field_name is not in self.CACHED_FIELDS
+        """
+        field_name, field_value = next(iter(kwargs.items()))
```

Comment:
What about enforcing that `kwargs` contains exactly one entry?
```suggestion
        [[field_name, field_value]] = kwargs.items()
```